jwst/extract_1d/extract.py

```diff
@@ -2863,14 +2863,18 @@ def do_extract1d(
         else:
             pixel_solid_angle = 1.  # not needed
 
-        source_type = input_model.meta.target.source_type
+        if isinstance(input_model, datamodels.SlitModel):
```

Comment:
This is the critical change that correctly populates `source_type` when the input is a `SlitModel`. All other changes below are fluff.